recipes/sol2/3.x.x/test_package/CMakeLists.txt

```
@@ -0,0 +1,16 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+file(GLOB SOURCE_FILES *.cpp)
+
+add_executable(${PROJECT_NAME} ${SOURCE_FILES})
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
+TARGET_COMPILE_DEFINITIONS (${PROJECT_NAME} PUBLIC -DCOMPILE_AS_CPP )
```

Should this option be added as an option to the `sol2` recipe itself?

```suggestion
target_compile_definitions(${PROJECT_NAME} PUBLIC COMPILE_AS_CPP)
```

---

Probably, yes!

---

you mean the `COMPILE_AS_CPP` ? Is this even needed? 
I wonder if not sol2 should add it if its compiled with `SOL_USING_CXX_LUA` for us , 

but I can add it after querying the lua option if needed, please advice !